agent/engine/docker_image_manager_integ_test.go

```diff
@@ -230,9 +231,10 @@ func TestIntegImageCleanupThreshold(t *testing.T) {
 
 	// Verify Task is stopped
 	verifyTaskIsStopped(taskEvents, testTask)
+	testTask.SetSentStatus(api.TaskStopped)
 
 	// Allow Task cleanup to occur
-	time.Sleep(2 * time.Second)
+	time.Sleep(5 * time.Second)
```

Comment:
Why `time.Sleep`? Is there an event we can listen for instead of setting an arbitrary time?

Is there a best practice around testing these sorts of interactions in Go?